### PR TITLE
Implement IP address data-source

### PR DESCRIPTION
**Pull request by marioreggiori**
### Description

The addition of the anxcloud_ip_address data-source allows users to retrieve more information about the requested resource (e.g. related prefix & vlan). 

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAnxCloudIPAddressDataSource'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccAnxCloudIPAddressDataSource -timeout 120m
?       github.com/anexia-it/terraform-provider-anxcloud        [no test files]
2022/05/20 15:36:02 Setting up new test environment
2022/05/20 15:36:02 Random Test Name: empty-river
=== RUN   TestAccAnxCloudIPAddressDataSource
--- PASS: TestAccAnxCloudIPAddressDataSource (36.37s)
PASS
ok      github.com/anexia-it/terraform-provider-anxcloud/anxcloud       71.417s
?       github.com/anexia-it/terraform-provider-anxcloud/anxcloud/testutils/environment [no test files]
?       github.com/anexia-it/terraform-provider-anxcloud/anxcloud/testutils/recorder    [no test files]
```

### Release Note
Release note for [CHANGELOG](https://github.com/anexia-it/terraform-provider-anxcloud/blob/master/CHANGELOG.md):

```release-note
* data-source/anxcloud_ip_address - added
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: tools/go.mod [hunk 1]
```diff
@@ -4,11 +4,12 @@ go 1.18
 
 require (
 	github.com/client9/misspell v0.3.4
-	github.com/golangci/golangci-lint v1.44.0
+	github.com/golangci/golangci-lint v1.46.2
```
[the remark this answers is not in the document]
Comment (marioreggiori) [reply tools/go.mod:7]: I'll extract it into a separate PR

File: docs/data-sources/ip_address.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  Retrieves an IP address.
+---
+
+# anxcloud_ip_address (Data Source)
+
+Retrieves an IP address.
+
+### Known limitations
+
+- Only unique IP addresses can be retrieved.
```
Comment (toothstone) [inline docs/data-sources/ip_address.md:15]: I think we need to expand a little on how 'uniqueness' is determined in this context, I assume this is based on the Engine context? So RFC 1918 IP addresses work fine, as long as they only exist once within the access scope of the Engine token used? From the Engine PoV ever 'IP address' object is unique, so we need to use concise language here.

File: tools/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (marioreggiori) [reply tools/go.mod:7]: #93

File: docs/data-sources/ip_address.md (hunk 2 above)
Comment (marioreggiori) [reply docs/data-sources/ip_address.md:15]: Something like this?
> Only IP addresses unique to the scope of your access token for Anexia Cloud can be retrieved.

Comment (toothstone) [reply docs/data-sources/ip_address.md]: Sounds good, but now I have another questions :sweat_smile:
Does this limitation apply to all retrieval styles? Can't I retrieve any address object via its identifier, isn't just the `address` search limited?

Comment (marioreggiori) [reply docs/data-sources/ip_address.md]: This should cover the limitation:
> When using the address argument, only IP addresses unique to the scope of your access token for Anexia Cloud can be retrieved. You can however get a unique result by specifying the related VLAN or network prefix.

Comment (toothstone) [reply docs/data-sources/ip_address.md]: Alright, that is much more helpful I think :)

File: anxcloud/data_source_ip_address_test.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+			// Get IPv6 address by it's shortened "address"
+			{
+				Config: `
+				data "anxcloud_ip_address" "test_v6_short" {
+					address = "fd00:0:2::2"
+				}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.anxcloud_ip_address.test_v6_short", "network_prefix_id", "a1fcffb46c034178bec553289a5951b2"),
```
Comment (LittleFox94) [inline anxcloud/data_source_ip_address_test.go:111]: why is this network prefix ID hardcoded?

File: anxcloud/data_source_ip_address_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+			// Get IPv6 address by it's expanded "address"
+			{
+				Config: `
+				data "anxcloud_ip_address" "test_v6_expanded" {
+					address = "FD00:0000:0002:0000:0000:0000:0000:0002"
+				}
+				`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.anxcloud_ip_address.test_v6_expanded", "network_prefix_id", "a1fcffb46c034178bec553289a5951b2"),
```
Comment (LittleFox94) [inline anxcloud/data_source_ip_address_test.go:123]: again, why is this network prefix ID hardcoded?

File: examples/data-sources/anxcloud_ip_address/data-source.tf [hunk 5]
```diff
@@ -0,0 +1,3 @@
+data "anxcloud_ip_address" "example" {
+  address = "10.244.2.50"
+}
```
Comment (LittleFox94) [inline examples/data-sources/anxcloud_ip_address/data-source.tf:3]: do we want to add an example where this is useful?

**Review by LittleFox94 — COMMENTED** [review]
some questions

File: anxcloud/data_source_ip_address_test.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+					address = "1.1.1.1.1"
+				}
+				`,
+				ExpectError: regexp.MustCompile("Failed to parse IP address."),
+			},
+		},
+	})
+}
+
+func testAccAnxCloudIPAddressDataSourceResources(envInfo environment.Info) string {
```
Comment (LittleFox94) [inline anxcloud/data_source_ip_address_test.go:120]: I like the idea, but this way we are probably creating all kinds those test IPs and prefixes for each of the tests, right?

Comment (marioreggiori) [reply anxcloud/data_source_ip_address_test.go:120]: Only once per `TestAccAnxCloudIPAddressDataSource` testrun. The test steps are incremental.

Comment (LittleFox94) [reply anxcloud/data_source_ip_address_test.go:120]: alright then :)

lgtm

**Pull request merged**